checker/manual/external-tools.tex

```diff
@@ -324,60 +324,90 @@ \section{Maven\label{maven}}
+
+First, create the property in the \code{properties} section of the POM:
 
 \begin{alltt}
-  <build>
-    <plugins>
-      <plugin>
-        <artifactId>maven-compiler-plugin</artifactId>
-        <configuration>
-          <source>1.8</source>
-          <target>1.8</target>
-          <fork>true</fork>
-          <executable>${env.CHECKERFRAMEWORK}/checker/bin/javac_maven</executable>
-        </configuration>
-      </plugin>
-    </plugins>
-  </build>
+<properties>
+    <!-- This property will be set by the Maven Dependency plugin -->
+    <annotatedJdk>${org.checkerframework:jdk8:jar}</annotatedJdk>
+</properties>
 \end{alltt}
 
-On Windows, the \code{javac\_maven.bat} script is automatically used instead
-of the \code{javac\_maven} script -- it is not necessary to include the \code{.bat}
-extension in the absolute path above.
+Change the reference to the \code{maven-dependency-plugin} within the \code{<plugins>}
+section, or add it if it is not present.
 
-This script assumes that the Checker Framework is the only annotation processor being run.
-If this is not the case, please modify the \code{javac\_maven} script accordingly.
+\begin{alltt}
+    <plugin>
+        <!-- This plugin will set properties values using dependency information -->
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <version>2.3</version>
+        <executions>
+            <execution>
+                <goals>
+                    <goal>properties</goal>
+                </goals>
+            </execution>
+        </executions>
+    </plugin>
+\end{alltt}
 
-\item Create a text file named \code{argfile} in the same directory as the \code{pom.xml}
-and include in it the command-line parameters to pass to the Java compiler.
 
-Example contents of \code{argfile}:
+\item Direct the Maven compiler plugin to use the desired checkers.
+Change the reference to the \code{maven-compiler-plugin} within the \code{<plugins>}
+section, or add it if it is not present.
+
+For example, to use the \code{org.checkerframework.checker.nullness.NullnessChecker}:
 
 \begin{alltt}
--processor org.checkerframework.checker.nullness.NullnessChecker
--AsuppressWarnings=purity.invalid.overriding
--Alint
--AprintErrorStack
--Awarns
--Xmaxwarns 10000
+    <plugin>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <configuration>
+            <source>1.8</source>
+            <target>1.8</target>
+            <fork>true</fork>
+            <annotationProcessors>
+                <!-- Add all the checkers you want to enable here -->
+                <annotationProcessor>org.checkerframework.checker.nullness.NullnessChecker</annotationProcessor>
+            </annotationProcessors>
+            <compilerArgs>
+                <!-- location of the annotated JDK, which comes from a Maven dependency -->
+                <arg>-Xbootclasspath/p:${annotatedJdk}</arg>
```

Comment:
Need to write \{ and \} instead of {  }, otherwise the braces do not show up in the built manual.